src/tests/Microsoft.PowerFx.Json.Tests/JSONTests.cs

```diff
@@ -36,5 +37,122 @@ public void Json_IncludeBinaryData_NoSideEffects()
             Assert.False(result.IsSuccess);
             Assert.Equal("The JSON function cannot serialize binary data in non-behavioral expression.", result.Errors.First().Message);
         }
+
+        [Fact]
+        public void Json_IncludeBinaryData_WithLazyRecord()
+        {
+            var config = new PowerFxConfig(Features.None);
+            config.EnableJsonFunctions();
+
+            var engine = new RecalcEngine(config);
+
+            var record = RecordType.Empty();
+            record = record.Add("Property", new LazyRecordType());
+
+            var formulaParams = RecordType.Empty();
+            formulaParams = formulaParams.Add("Var", record);
+
+            var result = engine.Check("JSON(Var)", formulaParams);
+
+            Assert.False(result.IsSuccess);
+            Assert.Equal("The JSON function cannot serialize tables / objects with a nested property called 'Property' of type 'Record'.", result.Errors.First().Message);
+        }
+
+        [Fact]
+        public void Json_IncludeBinaryData_WithLazyRecordAndFeature()
+        {
+            var config = new PowerFxConfig(Features.PowerFxV1);
+            config.EnableJsonFunctions();
+
+            var engine = new RecalcEngine(config);
+
+            var record = RecordType.Empty();
+            record = record.Add("Property", new LazyRecordType());
+
+            var formulaParams = RecordType.Empty();
+            formulaParams = formulaParams.Add("Var", record);
+
+            var result = engine.Check("JSON(Var)", formulaParams);
+
+            Assert.True(result.IsSuccess);
+        }
```

Comment:
Can you add a test where we actually check the result of the JSON function, to make sure that we not only compile, but we properly serialize lazy records? Also a test with lazy tables would be good.